File: check_avgs.py
import glob

files = glob.glob('src/app/**/*.tsx', recursive=True)
for f in files:
    with open(f, 'r', encoding='utf-8') as file:
        lines = file.readlines()
        for i, line in enumerate(lines):
            if '.reduce((s, v) => s + v, 0) /' in line or '.reduce((s, a) => s + (a.achievementPercentage || 0), 0) /' in line:
                print(f"--- {f}:{i+1} ---")
                start = max(0, i-5)
                end = min(len(lines), i+6)
                for j in range(start, end):
                    print(lines[j].rstrip())

File: check_avgs.js
const glob = require('glob');
const fs = require('fs');

const files = glob.sync('src/app/**/*.tsx');
for (const f of files) {
    const lines = fs.readFileSync(f, 'utf-8').split('\n');
    for (let i = 0; i < lines.length; i++) {
        if (lines[i].includes('.reduce((s, v) => s + v, 0) /') || lines[i].includes('.reduce((s, a) => s + (a.achievementPercentage || 0), 0) /')) {
            console.log(`--- ${f}:${i+1} ---`);
            const start = Math.max(0, i - 5);
            const end = Math.min(lines.length, i + 6);
            for (let j = start; j < end; j++) {
                console.log(lines[j].trimEnd());
            }
        }
    }
}
